pkg/deployer/lib/continuousreconcile/extension.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package continuousreconcile
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/robfig/cron/v3"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	duv1alpha1 "github.com/gardener/landscaper/apis/deployer/utils/continuousreconcile/v1alpha1"
+	"github.com/gardener/landscaper/pkg/deployer/lib/extension"
+)
+
+// ContinuousReconcileExtension returns an extension hook function which can handle continuous reconciliation.
+// It is meant to be used as a ShouldReconcile hook and might yield unexpected results if used with another hook handle.
+// The function which it takes as an argument is expected to take a time and return the time when the deploy item should be scheduled for the next automatic reconciliation.
+//   It should return nil if continuous reconciliation is not configured for the deploy item.
+func ContinuousReconcileExtension(nextReconcile func(context.Context, time.Time, *lsv1alpha1.DeployItem) (*time.Time, error)) extension.ReconcileExtensionHook {
+	return func(ctx context.Context, log logr.Logger, di *lsv1alpha1.DeployItem, target *lsv1alpha1.Target, hype extension.HookType) (*extension.HookResult, error) {
+		logger := log.WithName("continuousReconcileExtension")
+		logger.V(7).Info("execute")
+		if di == nil {
+			return nil, nil
+		}
+		nextRaw, err := nextReconcile(ctx, di.Status.LastReconcileTime.Time, di)
+		if err != nil {
+			return nil, fmt.Errorf("unable to check whether reconciliation is due: %w", err)
+		}
+		if nextRaw == nil {
+			logger.V(7).Info("no continuous reconcile specified")
+			return nil, nil
+		}
+		next := nextRaw.Round(time.Second)
+		now := time.Now().Round(time.Second)
+		res := &extension.HookResult{}
+		if next.After(now) {
+			// reconcile is not yet due
+			// since this is a ShouldReconcile hook, we can set AbortReconcile to false and the deploy item will still be reconciled if necessary (e.g. due to spec changes)
+			res.AbortReconcile = true
```

Comment:
can't we just directly return here as we are not due?

Reply:
No, we still need to compute the time when the continuous reconciliation is expected to be due and return a corresponding `RequeueAfter` value ...